wrapanapi/containers/providers/rhopenshift.py

```diff
@@ -0,0 +1,874 @@
+        Return: data if entity was created w/o errors
+        """
+        secret = self.kclient.V1Secret(**kwargs)
+        secret_name = secret.to_dict()['metadata']['name']
+        self.logger.info("creating secret %s", secret_name)
+        output = self.k_api.create_namespaced_secret(namespace=namespace, body=secret)
+        self.does_secret_exist(namespace=namespace, name=secret_name)
+        return output
+
+    def create_deployment_config(self, namespace, **kwargs):
+        """Creates Deployment Config entity using REST API.
+
+        Args:
+            namespace: openshift namespace where entity has to be created
+            kwargs: Deployment Config data
+        Return: data if entity was created w/o errors
+        """
+        dc = self.ociclient.V1DeploymentConfig(**kwargs)
+        dc_name = dc.to_dict()['metadata']['name']
+        self.logger.info("creating deployment config %s", dc_name)
+        output = self.o_api.create_namespaced_deployment_config(namespace=namespace, body=dc)
+        self.does_deployment_config_exist(namespace=namespace,
+                                          name=dc_name)
+        return output
+
+    def create_persistent_volume_claim(self, namespace, **kwargs):
+        """Creates Persistent Volume Claim entity using REST API.
+
+        Args:
+            namespace: openshift namespace where entity has to be created
+            kwargs: Persistent Volume Claim data
+        Return: data if entity was created w/o errors
+        """
+        pv_claim = self.kclient.V1PersistentVolumeClaim(**kwargs)
+        pv_claim_name = pv_claim.to_dict()['metadata']['name']
+        self.logger.info("creating persistent volume claim %s", pv_claim_name)
+        output = self.k_api.create_namespaced_persistent_volume_claim(namespace=namespace,
+                                                                      body=pv_claim)
+        self.does_persistent_volume_claim_exist(namespace=namespace,
+                                                name=pv_claim_name)
+        return output
+
+    def create_project(self, name):
+        """Creates Project(namespace) using REST API.
+
+        Args:
+            name: openshift namespace name
+        Return: data if entity was created w/o errors
+        """
+        proj = self.ociclient.V1Project()
+        proj.metadata = {'name': name}
+        self.logger.info("creating new project with name {n}".format(n=name))
+        output = self.o_api.create_project(body=proj)
+        self.does_project_exist(name=name)
+        return output
+
+    def does_project_exist(self, name, wait=5):
+        """Checks whether Project exists.
+
+        Args:
+            name: openshift namespace name
+            wait: entity should appear for this time then - True, otherwise False
+        Return: True/False
+        """
+        return wait_for(self._does_exist, num_sec=wait,
+                        func_kwargs={'func': self.o_api.read_project, 'name': name})[0]
```

Comment:
**QUESTION**: Shouldn't this be more like `wait_project_exists`? I would expect `does_$thing_exist` to immediately return the current state.

Reply:
usually, entity creation takes some time in openshift. So, I created all such methods with wait by default. However, I can rename them if you think it makes sense.

Reply:
already renamed

Reply:
Yeah, it would be ideal if methods that start with `do(es)_` did not wait and just tell you if it exists or not and leave the waiting for the `wait_` methods :)

Reply:
I already fixed this